corokafka/corokafka_header_pack.cpp

```diff
@@ -55,38 +56,46 @@ HeaderPack& HeaderPack::pop_back() {
     return *this;
 }
 
-void HeaderPack::erase(const std::string& name) {
-    _headers.erase(std::remove_if(_headers.begin(), _headers.end(), [&name](const HeaderNode& entry)->bool {
-        return StringEqualCompare()(entry.first, name);
+void HeaderPack::erase(const std::string& name, size_t relativePosition) {
+    size_t pos = 1;
+    _headers.erase(std::remove_if(_headers.begin(), _headers.end(), [&](const HeaderNode& header)->bool {
+        if (StringEqualCompare()(header.first, name)) {
+            return (relativePosition == 0) || (pos++ == relativePosition);
+        }
+        return false;
     }), _headers.end());
 }
 
-size_t HeaderPack::size() const {
+void HeaderPack::erase(const HeaderPack::ListType::iterator& it) {
+    _headers.erase(it);
```

Comment:
can we check if this iterator is valid here. at least check it is not end() or past it